lib/internal.js

```diff
@@ -193,7 +193,9 @@ export function resolve (obj, selector, deepFlag = false) {
         return value
       }
 
-      value = value.map((item) => resolve(item, names[i], true))
+      value = value.map(
+        (item) => resolve(item, names[i], true)
+      ).filter(item => item)
```

Comment:
This will filter out all values considered false by Javascript.

The output for now should have the empty values represented in the array without issue so that operators can treat them as they see fit.

For example, this test will return incorrect output
```js
_.computeValue({l1: {l2: [{l3: [{ a: 666}, {a: false}, {a: undefined}]}, {l4: 'some prop'}, {a: false}]}}, '$l1.l2.l3.a')
// expect: [666, false, undefined]
// actual: [666]
```

The fix for could be tricky. I think a large body of test will help here.